flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/FailureHandlingResult.java

```diff
@@ -38,6 +42,12 @@
     /** Delay before the restarting can be conducted. */
     private final long restartDelayMS;
 
+    /**
+     * The ExecutionVertexID refering to the ExecutionVertex the failure is originating from or
+     * {@code null} if it's a global failure.
+     */
+    @Nullable private final ExecutionVertexID failingExecutionVertexId;
+
     /** Reason why the failure is not recoverable. */
```

Comment:
The JavaDoc seems wrong. If I am not mistaken, then we will also set this field if the failure is recoverable, right?